api/src/pcapi/core/bookings/api.py

```diff
@@ -319,8 +319,10 @@ def cancel_booking_for_fraud(booking: Booking) -> None:
     validation.check_booking_can_be_cancelled(booking)
     _cancel_booking(booking, BookingCancellationReasons.FRAUD)
     logger.info("Cancelled booking for fraud reason", extra={"booking": booking.id})
-
-    if not transactional_mails.send_booking_cancellation_emails_to_user_and_offerer(booking, booking.cancellationReason):  # type: ignore [arg-type]
+    assert booking.cancellationReason is not None
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏Si par hasard `cancel_booking` fail, il y a une chance que `booking.cancellationReason` soit `None`
J'aurais plutôt changé la signature de `send_booking_cancellation_emails_to_user_and_offerer` pour accepter que `reason` puisse être `None`

```python
def send_booking_cancellation_emails_to_user_and_offerer(
    booking: Booking,
    reason: BookingCancellationReasons | None,
) -> bool:
```

Reply:
Yes c'est plus clean effectivement
Je me disait que si cancellationReason est None, on envoie pas le mail (de par le fonctionnement de la fonction d'envoi d'email), mais ta proposition est plus claire !

Reply:
je pense qu'il faut garder le assert pour mypy car si cancelled est vrai celà ne dit pas à mypy que cancellationReason n'est pas none